scripts/generate_visualc_files.pl

```diff
@@ -18,10 +18,20 @@
 my $vsx_sln_tpl_file = "scripts/data_files/vs2010-sln-template.sln";
 my $vsx_sln_file = "$vsx_dir/mbedTLS.sln";
 
+my $include_crypto = 0;
+if( @ARGV ) {
+    die "Invalid number of arguments" if scalar @ARGV != 1;
+    ($include_crypto) = @ARGV;
+}
+
 my $programs_dir = 'programs';
 my $header_dir = 'include/mbedtls';
 my $source_dir = 'library';
 
+if( $include_crypto ) {
+    $source_dir = 'crypto/library';
```

Comment:
No. If `$include_crypto`, VS must be told to compile files from both `crypto/library` and `library`.

Reply:
Hmm, I don't think I understand what would that do - why would we want to compile both sources at the same time, in the same project? I thought that the idea is that we can build either the submodule, or mbedTLS sources, both with mbedTLS headers.

Reply:
If we use the submodule, then crypto sources come from `crypto/library` and X509/TLS sources come from `library`.

Reply:
So should the crypto sources shadow mbedTLS ones, just as in the generate_errors scripts? We would then take all of the sources from crypto and non-duplicated files from mbedtls.

Reply:
Right.